### PR TITLE
Fix setting names on gateways

### LINKED ISSUE #21938: istioctl does not create correctly named gateways
```bash
$ cat dedicated-gateway.yaml 
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  profile: empty
  components:
    ingressGateways:
      - namespace: istio-system
        name: istio-ingressgateway
        enabled: false
      - namespace: istio-ingress-ns
        name: istio-ingress-1
        enabled: true

$ istioctl manifest generate -f dedicated-gateway.yaml | grep "^kind: Deployment" -A6
proto: tag has too few fields: "-"
kind: Deployment
metadata:
  name: istio-ingressgateway
  namespace: istio-ingress-ns
  labels:
    app: istio-ingressgateway
    istio: ingressgateway
```
Other k8s resources are similarly misnamed. Expected results is that the name of the gateway resources is `istio-ingress-1`.

### PR DESCRIPTION
**Pull request by ostromart**
Fixes https://github.com/istio/istio/issues/21938.

### CODE REVIEW

File: operator/data/profiles/default.yaml [hunk 1]
```diff
@@ -451,6 +451,7 @@ spec:
         type: ClusterIP
         env:
           ISTIO_META_ROUTER_MODE: "sni-dnat"
+        name: istio-egressgateway
```
Comment (richardwxn) [inline operator/data/profiles/default.yaml]: Is this addition necessary? just try to reduce the overlap of the settings in values and in components spec

**Review by richardwxn — COMMENTED** [review]
just one comment,  others LGTM

File: operator/data/profiles/default.yaml (hunk 1 above)
Comment (ostromart) [reply operator/data/profiles/default.yaml]: yes, it's required because i had to make changes in the charts to not hard code the gateway name. in the case where the name is not supplied in IOP, we need some default here.

Comment (richardwxn) [reply operator/data/profiles/default.yaml]: oh ok. Also I think the other issues complaining unable to set user defined gateways are caused by this as well?

Comment (ostromart) [reply operator/data/profiles/default.yaml]: right

**Review by howardjohn — APPROVED** [review]

**Review by richardwxn — APPROVED** [review]

**Review by myidpt — APPROVED** [review]

File: tests/integration/security/main_test.go [hunk 2]
```diff
@@ -60,5 +60,6 @@ func setupConfig(cfg *istio.Config) {
 components:
   egressGateways:
   - enabled: true
+    name: istio-egressgateway
```
Comment (howardjohn) [inline tests/integration/security/main_test.go]: Is this required, or just "best practice"?

Comment (ostromart) [reply tests/integration/security/main_test.go]: it's required in principle and will also be in practice when we add validation for it.

**Comment by howardjohn** [comment]
What is the point of all the `| default`s then?
